scripts/updateDatabase.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+## This script will create or replace a database using a mysql dump file. It takes the DB name (-d or --database) and dump filepath (-f or --file) as arguments.
+## A config file populated with mySQL username, password and host is also required. Its filepath can be optionally specified (-c or --config), otherwise a default 'config.properties' file is assumed.
+## author: jcook
+
+dbFilepath=
+dbName=
+configFilepath=config.properties
+## Parse command-line arguments
+while (( "$#" )); do
+	case "$1" in
+		-f|--file)
+			dbFilepath=$2;
+			shift 2;
+			;;
+		-d|--database)
+			dbName=$2;
+			shift 2;
+			;;
+		-c|--config)
+			configFilepath=$2;
+			shift 2;
+			;;
+		-*|--*=)
+			echo "Error: Unsupported flag $1"
+			exit 1
+	esac
+done
+
+## If missing arguments, explain usage
+if [ -z "$dbFilepath" ] || [ -z "$dbName" ]
+then
+	echo "Create or replace mySQL databases";
+	echo "Usage: bash updateDatabase.sh -f databaseFile -d databaseName [-c configFile] ";
+	exit 1
+fi
+
+## Parse config file for username and password properties
+echo "Reading $configFilepath";
+username=
+password=
+host=localhost
+port=3306
+while read line; do
+	if [[ $line == username* ]]
+ 	then
+ 		username=${line#*=}
```

Comment:
Just curious, what does the #*= do?

Reply:
It is a weird one: https://stackoverflow.com/questions/41806725/what-is-the-meaning-of-1-in-bash

As I interpret it, the expression is three  parts, the `#`, `*` and `{}`. Together, they are saying find the pattern matching the `=`, trim anything preceding it and capture the rest. I think the `{}` surrounding the whole expression is what captures the remainder of the string. The 'drop anything preceding it' is apparently due to the `#` combined with the `*`. More technically, the # means drop _shortest_ preceding string, while having the operator `##` would mean drop the _longest_ preceding string.

Reply:
The expected string by the way is username=mySQLUsername